module/input/Camera/src/Camera.cpp

```diff
@@ -462,12 +228,408 @@ namespace module::input {
 
                 // Put the buffer back on the queue and show an error message
                 arv_stream_push_buffer(stream, buffer);
-                NUClear::log<NUClear::WARN>("Failed getting buffer from", context->name, "with error", msg);
+                reactor.log<NUClear::WARN>(
+                    fmt::format("Failed getting buffer from the {} camera with error {}", context->name, msg));
             }
         }
     }
 
-    void Camera::control_lost(ArvGvDevice*, CameraContext* context) {
-        NUClear::log<NUClear::FATAL>(fmt::format("Control of a the {} camera has been lost", context->name));
+    void Camera::control_lost(ArvGvDevice* /*device*/, CameraContext* context) {
+        context->reactor.log<NUClear::FATAL>(fmt::format("Control of the {} camera has been lost", context->name));
+    }
+
+    /************************************************************************
+     *   ____                _          ____                                *
+     *  / ___|_ __ ___  __ _| |_ ___   / ___|__ _ _ __ ___   ___ _ __ __ _  *
+     * | |   | '__/ _ \/ _` | __/ _ \ | |   / _` | '_ ` _ \ / _ \ '__/ _` | *
+     * | |___| | |  __/ (_| | ||  __/ | |__| (_| | | | | | |  __/ | | (_| | *
+     *  \____|_|  \___|\__,_|\__\___|  \____\__,_|_| |_| |_|\___|_|  \__,_| *
+     *                                                                      *
+     ************************************************************************/
+    CameraContext configure_camera(const Configuration& config, Camera& reactor) {
+        // Find the serial number for this camera
+        const auto& serial_number = config["serial_number"].as<std::string>();
+
+        // Strip the .yaml off the name of the file to get the name of the camera
+        std::string name = std::filesystem::path(config.path).stem();
+
+        // Find the camera with the correct serial number
+        auto find_camera = [](const std::string& name, const std::string& serial_number) {
+            unsigned int devices = arv_get_n_devices();
+            for (unsigned int i = 0; i < devices; ++i) {
+                if (serial_number == arv_get_device_serial_nbr(i)) {
+                    return i;
+                }
+            }
+            throw std::runtime_error(fmt::format("{} camera with serial number {} not found", name, serial_number));
+        };
+
+        // Find the device, and if we can't check for new devices
+        int device_no = 0;
+        try {
+            device_no = find_camera(name, serial_number);
+        }
+        catch (const std::runtime_error&) {
+            arv_update_device_list();
```

Comment:
Is this rare enough that the try-catch is more performant?

Reply:
Yeah this should only happen when a camera disappears/unplugged etc and needs to be rediscovered